### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by harshitajain06**
- Implemented classes and objects in Ruby.
- Implemented encapsulation and inheritance with Ruby.
- Run the program using the command line.

### CODE REVIEW

**Comment by TedLivist** [comment]
## STATUS: INVALID CODE REVIEW REQUEST 🛑 🛑

Hi @harshitajain06,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### - Your README is a dummy file from the template. Kindly update your README to be descriptive
### - Kindly ensure your linter checks run for this PR

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: README.md [hunk 1]
```diff
@@ -36,244 +13,76 @@ After you're finished please remove all the comments and instructions!
...
-<!-- LICENSE -->
-
-## 📝 License <a name="license"></a>
-
-This project is [MIT](./LICENSE) licensed.
-
-_NOTE: we recommend using the [MIT license](https://choosealicense.com/licenses/mit/) - you can set it up quickly by [using templates available on GitHub](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-license-to-a-repository). You can also use [any other license](https://choosealicense.com/licenses/) if you wish._
-
-<p align="right">(<a href="#readme-top">back to top</a>)</p>
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
```
Comment (t-yanick) [inline README.md:96]: - [ ] Great job updating the README. Unfortunately, you are missing the `License` section. Kindly add it.

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
+class Person
+    attr_reader :id, :name, :age
+
```
Comment (t-yanick) [inline person.rb]: - [ ] You should create getter for name and age. You could to a shortcut and create setter and getter for name and age and then a setter just for ID. You can read this [lesson](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_principles_examples.md#useful-shortcuts-attr_reader-attr_writer-and-attr_accessor) for more details.
```ruby
  attr_accessor :name, :age 
  # *:id should only be able to be retrieved so only the Getter is defined
  attr_reader :id
```

File: student.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+require_relative 'person'
+class Student < Person
+    attr_accessor :classroom
```
Comment (t-yanick) [inline student.rb]: - [ ] `classroom` is not modified. As such you don't need to set it as a `setter` and `getter`. You can delete this line of code.
- You can create the intialize method as below:
```ruby
  def initialize(age, classroom, name = "Unknown", parent_permission = true)  
    super(age, name, parent_permission)
    @classroom = classroom
  end
```
- `Constructor extends parent's constructor by adding @classroom and a parameter for it.` You can [read more by clicking on this link](`https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/class_and_object_for_irb.md#constructor-and-instance-vars`)

File: teacher.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+
+
+
+
+
+  
+  
+  
+
```
Comment (t-yanick) [inline teacher.rb]: - [ ]Kindly delete these empty lines.

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED

Hi @harshitajain06 , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  All three .rb files added
:heavy_check_mark: Use of inheritance :+1: 

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AnselemOdims — APPROVED** [review]
### Hi  @harshitajain06 🙋‍♀️, 
Good job in making the required changes 👏 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Use the tag @AnselemOdims for a quicker reply


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**